### PR TITLE
add dotnet 5 azure function skeleton

### PR DESCRIPTION
**Pull request by cmgrd**
This PR adds dotnet 5 azure functions via a new MarketRoles solution.

The three new functions are:
- Ingestion
- Processing
- Outbox

New CI/CD GitHub action files are created for each of those.

Terraform files are added and updated. Also; old function .tf-files are deleted.

References:
Why: https://docs.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#why-net-isolated-process
How: https://github.com/Azure/azure-functions-dotnet-worker

### CODE REVIEW

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.TestTool/TestTool/Program.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace TestTool
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+
+            var connectionString = "Endpoint=sb://sbn-marketroles-endk-u.servicebus.windows.net/;SharedAccessKeyName=sbnar-marketroles-sender;SharedAccessKey=iiKzbA0xI60YeZXU/T8uZ5a32rwp9f0Flera0FAQaoM=";
```
Comment (erduna) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.TestTool/TestTool/Program.cs:14]: Should this be hardcoded?

**Review by erduna — APPROVED** [review]
see single comment

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.TestTool/TestTool/Program.cs (hunk 1 above)
Comment (cmgrd) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.TestTool/TestTool/Program.cs:27]: No.

File: .github/workflows/ingestion-cd.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: Ingestion CD
+
+on:
+  push:
+    branches:
+      - main
+      - cmgrd/dotnet-playground
```
Comment (Korgath) [inline .github/workflows/ingestion-cd.yml:20]: Should be removed before merging (this is in several files, but only commenting here)

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.EntryPoints.Outbox/ActorMessageDispatcher.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        public static void Run(
+            [TimerTrigger("%ACTOR_MESSAGE_DISPATCH_TRIGGER_TIMER%")] string timerInformation,
+            FunctionContext context)
+        {
+            var logger = context.GetLogger("ActorMessageDispatcher");
+            logger.LogInformation($"C# Timer trigger function executed at: {DateTimeOffset.UtcNow} (UTC)");
+            logger.LogInformation($"From function timer trigger input: {timerInformation}");
+        }
+    }
+}
```
Comment (Korgath) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.EntryPoints.Outbox/ActorMessageDispatcher.cs:32]: This ActorMessageDispatcher looks completely different from the current one in main which calls a forwardmessageService. Is this on purpose or is it to be implemented again later?

Comment (cmgrd) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.EntryPoints.Outbox/ActorMessageDispatcher.cs:32]: This is just an empty skeleton for demonstration purposes.

File: .github/workflows/ingestion-cd.yml (hunk 2 above)
Comment (cmgrd) [reply .github/workflows/ingestion-cd.yml]: Removed.

**Review by Korgath — APPROVED** [review]

**Review by MrDach — APPROVED** [review]
👍

**Pull request merged**